### ISSUE TITLE
context (region) at definition persistently alters evaluation of abstract variables

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1572]
**Reported by andrew wittenberg on 11 Jun 2015 15:52 UTC**
Set a region, then list a couple of variables:

```
 	NOAA/PMEL TMAP
 	FERRET v6.95 (beta)
 	Linux 2.6.32-504.16.2.el6.x86_64 64-bit - 06/09/15
 	11-Jun-15 11:41     

yes? set reg/x=10:20
yes? let a = {1,2,3}
yes? list a, a[Column  1: A[X=0.5:3.5](i=@shf]
) is {1,2,3}
 Column  2: A[is {1,2,3}
             A  A
1   / 1:  1.000..
2   / 2:  2.000..
3   / 3:  3.000..
```

Presumably the second column doesn't appear because the region is outside the dimensional bounds of the variable.  (Need to fix the display formatting here though, as the "...." in the second column is butting up against the values in the first column.)

Now cancel the region and try again:

```
yes? can reg/x
yes? list a, a[i=@shf](X=10:20@SHF:1])
 Column  1: A[is {1,2,3}
 Column  2: A[X=10:20@SHF:1](X=0.5:3.5]) is {1,2,3}
             A  A
1   / 1:  1.000..
2   / 2:  2.000..
3   / 3:  3.000..
```

Same result!  Why?  It seems that something is stuck in the definitions, retaining a context that's no longer relevant.

Now redefine the first variable and try again:

```
yes? let a = {1,2,3}
yes? list a, a[            X: 0.5 to 3.5
 Column  1: A is {1,2,3}
 Column  2: A[X=@SHF:1](i=@shf]
) is {1,2,3} (shifted by 1 pts on X)
             A     A
1   / 1:  1.000  2.000
2   / 2:  2.000  3.000
3   / 3:  3.000   ....
```

Now we get what we expected.  What's going on here?

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2300

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 11 Jun 2015 17:46 UTC**
It seems to me that CANCEL REGION should do a CANCEL MEMORY, but it doesn't.  Listing just the a[ set reg/x=10:20
yes? let a = {1,2,3}
yes? list a
             VARIABLE : {1,2,3}
             SUBSET   : 3 points (X)
 1   / 1:  1.000
 2   / 2:  2.000
 3   / 3:  3.000
yes? list a[i=@shf](x=@shf],

```
yes?)
             VARIABLE : {1,2,3}
             X        : 10 to 20 (shifted by 1 pts)
        ....
yes? cancel region
yes? list a[            VARIABLE : {1,2,3}
             X        : 10 to 20 (shifted by 1 pts)
        ....
yes? can mem
 TEMPORARY data cleared from memory
yes? list a[i=@shf](i=@shf]
)
             VARIABLE : {1,2,3}
                        shifted by 1 pts on X
             SUBSET   : 3 points (X)
 1   / 1:  2.000
 2   / 2:  3.000
 3   / 3:   ....
```

( and yes, a formatting error on the listing when shown with variable a)

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 12 Jun 2015 19:15 UTC**
This is in fact related to #2301. Previously Ferret wouldn't compute the shifted variable; the result doesn't even have the right size.

```

    yes? set region/x=10:20
    yes? let a = {1,2,3}
    yes? list a
                 VARIABLE : {1,2,3}
                 SUBSET   : 3 points (X)
     1   / 1:  1.000
     2   / 2:  2.000
     3   / 3:  3.000
    yes? list a[                VARIABLE : {1,2,3}
                 X        : 10 to 20 (shifted by 1 pts)
            ....
```

With bounds-checking turned on the a[i=@shf](i=@shf]
) results in a Fortran runtime error, array index out of range. 

With the fix to #2301, a[is correctly returned. Because a is on an abstract axis, it can list a and a[i=@shf](i=@shf]) whether there is a SET REGION in effect or not.


```
yes? set reg/x=10:20
yes? let a = {1,2,3}
yes? list a, a[Column  1: A[X=0.5:3.5](i=@shf]
) is {1,2,3}
 Column  2: A[is {1,2,3} (shifted by 1 pts on X)
             A     A
1   / 1:  1.000  2.000
2   / 2:  2.000  3.000
3   / 3:  3.000   ....
yes? can reg/x
yes? list a, a[i=@shf](X=10:20@SHF:1])
             X: 0.5 to 3.5
 Column  1: A is {1,2,3}
 Column  2: A[X=@SHF:1] is {1,2,3} (shifted by 1 pts on X)
             A     A
1   / 1:  1.000  2.000
2   / 2:  2.000  3.000
3   / 3:  3.000   ...
```
.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 14 Jun 2015 00:22 UTC**
This fix also appears to have resolved #1462.
